src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3360,9 +3373,9 @@ private BoundExpression BindNullCoalescingOperator(BinaryExpressionSyntax node,
                     Conversion.NoConversion, CreateErrorType(), hasErrors: true);
             }
 
-            if (leftOperand.IsLiteralDefault())
+            if (leftOperand.IsLiteralDefault() || leftOperand.IsTypelessNew())
```

Comment:
>leftOperand.IsLiteralDefault() || leftOperand.IsTypelessNew() [](start = 16, length = 61)

Please add a PROTOTYPE comment to track that this test should be `leftOperand.Type is null`.